packages/deployer/src/deploySolutionItems.ts

```diff
@@ -114,9 +114,10 @@ export function deploySolutionItems(
       templates = _evaluateSharedViewSources(templates);
     }
 
-    // Create an ordered graph of the templates so that dependencies are created
-    // before the items that need them
-    const cloneOrderChecklist: string[] = common.topologicallySortItems(
+    // Create an ordered graph of the templates so that dependencies are created before the items that need them.
+    // Because cycles are permitted, we also keep track of items that need to be patched later because their
+    // dependencies are necessarily created after they are created.
+    const { buildOrder, itemsToBePatched } = common.topologicallySortItems(
```

Comment:
@MikeTschudi should missingDependencies be in here also?

Reply:
It's not used for deployment at this time, but I left it in `topologicallySortItems()` for testing.